src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5033,38 +5032,41 @@ void SessionImpl::loadCategories()
 {
     m_categories.clear();
 
-    QFile confFile {(specialFolderLocation(SpecialFolder::Config) / CATEGORIES_FILE_NAME).data()};
-    if (!confFile.exists())
-    {
-        // TODO: Remove the following upgrade code in v4.5
-        // == BEGIN UPGRADE CODE ==
-        upgradeCategories();
-        m_needUpgradeDownloadPath = true;
-        // == END UPGRADE CODE ==
-
-//        return;
-    }
+    const int fileMaxSize = 1024 * 1024;
+    const Path path = specialFolderLocation(SpecialFolder::Config) / CATEGORIES_FILE_NAME;
 
-    if (!confFile.open(QFile::ReadOnly))
+    const auto readResult = Utils::IO::readFile(path, fileMaxSize);
+    if (!readResult)
     {
-        LogMsg(tr("Failed to load Categories. File: \"%1\". Error: \"%2\"")
-               .arg(confFile.fileName(), confFile.errorString()), Log::CRITICAL);
-        return;
+        if (readResult.error().status == Utils::IO::ReadError::NotExist)
```

Comment:
Damn it. Your new helper has confused all the cards in this particular place... it still breaks the previous algorithm.
It should read the "categories" file after the upgrade is done.

Reply:
sorry, fixed.